src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java

```diff
@@ -229,6 +236,148 @@ public Mono<TransactionInfoDto> getTransactionInfo(
+            return switch (baseTransactionView) {
+                case Transaction transaction ->
+                        evaluateOutcome(transaction.getStatus(), transaction.getSendPaymentResultOutcome(), transaction.getPaymentGateway(), transaction.getGatewayAuthorizationStatus(), transaction.getAuthorizationCode(), transaction.getAuthorizationErrorCode())
+                                .fold(_voidNull -> new TransactionOutcomeInfoDto(),
+                                        outcomeEnum -> new TransactionOutcomeInfoDto()
+                                        .outcome(outcomeEnum)
+                                        .totalAmount(outcomeEnum == TransactionOutcomeInfoDto.OutcomeEnum.NUMBER_0 ? transaction.getPaymentNotices().stream().mapToInt(it.pagopa.ecommerce.commons.documents.PaymentNotice::getAmount).sum() + Optional.ofNullable(transaction.getFeeTotal()).orElse(0) : null));
+                default -> throw new IllegalStateException("Unexpected value: " + baseTransactionView);
+            };
+    }
+
+    private Either<Void, TransactionOutcomeInfoDto.OutcomeEnum> evaluateOutcome(
+                                                                                TransactionStatusDto status,
+                                                                                TransactionUserReceiptData.Outcome sendPaymentResultOutcome,
+                                                                                String paymentGateway,
+                                                                                String gatewayAuthorizationStatus,
+                                                                                String authorizationCode,
+                                                                                String authorizationErrorCode
+    ) {
+        switch (status) {
+            case NOTIFIED_OK -> {
+                return Either.right(TransactionOutcomeInfoDto.OutcomeEnum.NUMBER_0);
+            }
+            case NOTIFICATION_REQUESTED, NOTIFICATION_ERROR -> {
+                return sendPaymentResultOutcome != null
+                        && sendPaymentResultOutcome.equals(TransactionUserReceiptData.Outcome.OK)
+                                ? Either.right(TransactionOutcomeInfoDto.OutcomeEnum.NUMBER_0)
+                                : Either.right(TransactionOutcomeInfoDto.OutcomeEnum.NUMBER_25);
+            }
+            case NOTIFIED_KO, REFUNDED -> {
+                return Either.right(TransactionOutcomeInfoDto.OutcomeEnum.NUMBER_25);
+            }
+            case EXPIRED_NOT_AUTHORIZED -> {
+                return Either.right(TransactionOutcomeInfoDto.OutcomeEnum.NUMBER_4);
+            }
+            case CANCELED, CANCELLATION_EXPIRED -> {
+                return Either.right(TransactionOutcomeInfoDto.OutcomeEnum.NUMBER_8);
+            }
+            case CLOSURE_ERROR, AUTHORIZATION_COMPLETED -> {
```

Comment:
This status (and the others below) are all transient statuses that can change dinamically.
The returned status in those cases is not a final one, meaning that caller have to poll to check if a final status have been reached.
Looking at the api how a caller can distinguish a final status from an intermediate one?

Reply:
You're right. Added final status flag and its evaluation here 3147f363366c39f5fe701ccaf4b185c967c5695d